iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -2169,6 +2161,12 @@ private List<WALFlushListener> logDeletionInWAL(
         walFlushListeners.add(walFlushListener);
       }
     }
+    // Some time the deletion operation doesn't have any related tsfile processor or memtable,
+    // but it's still necessary to write to the WAL, so that iotconsensus can synchronize the delete
+    // operation to other nodes.
+    if (walFlushListeners.isEmpty()) {
+      walFlushListeners.add(getWALNode().log(TsFileProcessor.MEMTABLE_NOT_EXIST, deleteDataNode));
+    }
```

Comment:
If config.getDataRegionConsensusProtocolClass() != IOT_CONSENSUS, there will be an UnsupportedOperationException thrown from getWALNode() method.

Reply:
Indeed. Now getWALNode() return Optional.